app/components/bookings/create/create-booking.component.ts

```diff
@@ -1,9 +1,130 @@
 import {Component} from '@angular/core';
+import {Workpackage} from "../../../models/workpackage/workpackage";
+import {NavParams, NavController, Alert} from "ionic-angular/index";
+import {BookingService} from "../../../services/bookings/booking.service";
+import {AbstractControl, ControlGroup, FormBuilder, Validators} from "@angular/common";
+import moment = require("moment/moment");
+import {BookingSessionService} from "../../../services/bookings/booking-session.service";
+import {BookingSession, Booking} from "../../../models/booking/booking";
+import Moment = moment.Moment;
+import {Response} from "@angular/http";
 
 @Component({
   templateUrl: 'build/components/bookings/create/create-booking.component.html'
 })
 export class CreateBookingComponent {
-  constructor() {
+  private _workpackage: Workpackage;
+  private _bookingSession: BookingSession;
+  effort : AbstractControl;
+  description: AbstractControl;
+  date: AbstractControl;
+  newEtc: AbstractControl;
+  authForm: ControlGroup;
+  myDate = moment().startOf('day').format('YYYY-MM-DD');
+  myEffort = moment().startOf('day').format('HH:mm');
+  myEtc: number;
+
+  constructor(navParams: NavParams,
+              private _bookingsService: BookingService,
+              private _bookingSessionService: BookingSessionService,
+              private _fb: FormBuilder,
+              private _nav: NavController) {
+
+    this._workpackage = navParams.get('workpackage');
+    this._bookingSession = navParams.get('bookingSession');
+
+    if(this._bookingSession) {
+      this.myEffort = this._calcEffortForSession(this._bookingSession.startTime, moment().unix());
+      this.refreshEtc();
+    }
+
+    this.authForm = _fb.group({
+      'description': ['', Validators.compose([Validators.required, Validators.minLength(1)])],
+      'date': ['', Validators.required],
+      'effort': ['', Validators.required]
+    });
+
+    this.newEtc = this.authForm.controls['newEtc'];
+    this.effort = this.authForm.controls['effort'];
+    this.description = this.authForm.controls['description'];
+    this.date = this.authForm.controls['date'];
+    this.myEtc = this._workpackage.etc;
+  }
+
+  onSubmit() {
+    if (this._checkFieldsSet()) {
+      var booking: Booking = new Booking();
+      booking.date = moment(this.date.value);
+      booking.description = this.description.value;
+      booking.effort = this._stringToWorkday(this.effort.value);
+      booking.workpackage = this._workpackage;
+      this._bookingsService.create(booking, this.myEtc).subscribe((returnedBooking: Booking) => {
+        let alert = Alert.create({
```

Comment:
I think using an alert here, which has to closed by pressing ok, is rather disruptive for the users workflow. I think just using a Toast while redirecting to the BookingOverviewComponent would be a more appropriate, of informing the user of what is going on.
Also what does happen, if something goes wrong (no internet connection, server unavailable etc..)? There should be some kind of error handling and feedback to the user.